contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,1212 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import "./System.sol";
+import "./lib/Utils.sol";
+import "./interface/IBSCValidatorSet.sol";
+import "./interface/ICrossChain.sol";
+import "./interface/IGovToken.sol";
+import "./interface/IStakeCredit.sol";
+import "./interface/ITokenHub.sol";
+import "./lib/RLPDecode.sol";
+
+contract StakeHub is System, Initializable {
+    using RLPDecode for *;
+    using Utils for string;
+    using Utils for bytes;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /*----------------- constants -----------------*/
+    uint256 private constant BLS_PUBKEY_LENGTH = 48;
+    uint256 private constant BLS_SIG_LENGTH = 96;
+
+    address public constant DEAD_ADDRESS = address(0xdEaD);
+    uint256 public constant LOCK_AMOUNT = 1 ether;
+    uint256 public constant REDELEGATE_FEE_RATE_BASE = 10000; // 100%
+
+    uint256 public constant BREATHE_BLOCK_INTERVAL = 1 days;
+
+    //TODO: set to the correct bytes of abi.encode({{BCExistingConsensusAddresses}}) and abi.encode({{BCExistingVoteAddresses}}) when landing on mainnet and testnet
+    // this will be set to proper value after the first sunset hardfork on Beacon Chain
+    bytes private constant INIT_BC_CONSENSUS_ADDRESSES =
+        hex"00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000";
+    bytes private constant INIT_BC_VOTE_ADDRESSES =
+        hex"00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000";
+
+    /*----------------- errors -----------------*/
+    // @notice signature: 0xd7485e8f
+    error StakeHubPaused();
+    // @notice signature: 0xb1d02c3d
+    error InBlackList();
+    // @notice signature: 0xf2771a99
+    error OnlyAssetProtector();
+    // @notice signature: 0x5f28f62b
+    error ValidatorExisted();
+    // @notice signature: 0xfdf4600b
+    error ValidatorNotExist();
```

Comment:
`ValidatorNotExist` -> `ValidatorNotExisted` to align to others.

Reply:
Done